fvm/environment/derived_data_invalidator.go

```diff
@@ -101,14 +111,34 @@ func (invalidator ProgramInvalidator) ShouldInvalidateEntry(
 	}
 
 	// invalidate all programs depending on any of the contracts that were
-	// updated.  A program has itself listed as a dependency, so that this
+	// updated. A program has itself listed as a dependency, so that this
 	// simpler.
-	for _, key := range invalidator.ContractUpdateKeys {
-		_, ok := program.Dependencies[key.Address]
+	for _, key := range invalidator.ContractUpdates.Updates {
+		loc := common.AddressLocation{
```

Comment:
recreating the address location object on each ShouldInvalidateEntry call is expensive.  maybe do some pre-processing.  similar below for common.Address

Reply:
I will address this in a separate PR when I switch everything to Address Location https://github.com/onflow/cadence/pull/2382